scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -57,11 +57,12 @@ def _highly_variable_genes_seurat_v3(
         )
 
     X = adata.layers[layer] if layer is not None else adata.X
-    if check_nonnegative_integers(X) is False:
-        raise ValueError(
-            "`pp.highly_variable_genes` with `flavor='seurat_v3'` expects "
-            "raw count data."
-        )
+    if check_values:
+        if check_nonnegative_integers(X) is False:
+            warnings.warn(
+                "`flavor='seurat_v3'` expects raw count data, but non-integers were found.",
+                UserWarning,
+            )
```

Comment:
```suggestion
    if check_values and (check_nonnegative_integers(X) == False):
        warnings.warn(
            "`flavor='seurat_v3'` expects raw count data, but non-integers were found.",
            UserWarning,
        )
```

* `is` to `==` since `is` can be weird (object identity, not value) 
* `and` since it's short-circuiting and reduces total indentation